online-inference/hf-llm/service/service.py

```diff
@@ -0,0 +1,140 @@
+    hf_id = '/'.join(path_uri.split("/")[-2:])
+    config = AutoConfig.from_pretrained(hf_id)
+    tokenizer = AutoTokenizer.from_pretrained(hf_id)
+
+    with tensorizer.utils.no_init_or_tensor():
+        model = AutoModelForCausalLM.from_config(config)
+
+    dtype_str = "/fp16" if args.precision == "float16" else ""
+    start = time.time()
+    load_artifact(f"{path_uri}{dtype_str}/model.tensors", model)
+    logger.info(
+        f"Model loaded successfully in {time.time() - start:.2f} seconds"
+    )
+
+    return TextGenerationPipeline(model=model, tokenizer=tokenizer, device=args.device_id)
+
+
+def load_model_local(path_uri: str) -> TextGenerationPipeline:
+    dtype = torch.float16 if args.precision == "float16" else torch.float32
+    model = AutoModelForCausalLM.from_pretrained(path_uri, torch_dtype=dtype)
+    tokenizer = AutoTokenizer.from_pretrained(path_uri)
+    return TextGenerationPipeline(model=model, tokenizer=tokenizer, device=args.device_id)
+
+
+def load_model(path_uri: str) -> TextGenerationPipeline:
+    logger.info(f"Loading {args.precision} model from: {path_uri}")
+    if path_uri.startswith("s3://"):
+        return load_model_s3(path_uri)
+    else:
+        return load_model_local(path_uri)
+
+
+class Completion(BaseModel):
+    prompt: str
+    max_new_tokens: Optional[int] = 10
+    temperature: Optional[float] = None
+    top_p: Optional[float] = None
+    top_k: Optional[int] = None
+    typical_p: Optional[float] = None
+    repetition_penalty: Optional[float] = None
+    do_sample: Optional[bool] = False
+    penalty_alpha: Optional[float] = None
+    num_return_sequences: Optional[int] = 1
+    stop_sequence: Optional[str] = None
+    bad_words: Optional[List] = None
+
+
+app = FastAPI(title="LLM Inference API")
+logger.info(f"Setting CUDA device ID: {args.device_id}")
+torch.cuda.set_device(args.device_id)
+pipe = load_model(args.model_uri)
+
+
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=["*"],
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+
+@app.get("/")
+def get_health():
+    logger.info("Health check requested")
+    return "OK"
+
+
+@app.post("/completion")
+async def completion(completion: Completion):
+    try:
+        output = pipe(
+            completion.prompt,
+            max_new_tokens=completion.max_new_tokens,
+            temperature=completion.temperature,
+            top_p=completion.top_p,
+            top_k=completion.top_k,
+            repetition_penalty=completion.repetition_penalty,
+            do_sample=completion.do_sample,
+            penalty_alpha=completion.penalty_alpha,
+            num_return_sequences=completion.num_return_sequences,
+            stop_sequence=completion.stop_sequence
+        )
+        return Response(content=json.dumps(output), media_type="application/json", status_code=200)
+    except Exception as e:
+        return Response(content=str(e), media_type="text/plain", status_code=500)
```

Comment:
You can provide more detail via the `logging` module while also preventing any information from leaking in server responses with something like this:
```suggestion
        import traceback

        logger.error(traceback.format_exc())
        return Response(
            content=(
                f"Server encountered an error of type {type(e).__name__}."
                "\nSee logs for details."
            ),
            media_type="text/plain",
            status_code=500,
        )
```